Sources/XcodeGraphMapper/Mappers/Project/PBXProjectMapper.swift

```diff
@@ -45,8 +45,15 @@ struct PBXProjectMapper: PBXProjectMapping {
 
         // Map PBXTargets to domain Targets
         let targetMapper = PBXTargetMapper()
-        let targets = try await pbxProject.targets.serialCompactMap {
-            try await targetMapper.map(pbxTarget: $0, xcodeProj: xcodeProj)
+        let targets = try await withThrowingTaskGroup(of: Void.self, returning: [Target].self) { taskGroup in
+            var targets: [Target] = []
+            for pbxTarget in pbxProject.targets {
+                taskGroup.addTask {
+                    targets.append(try await targetMapper.map(pbxTarget: pbxTarget, xcodeProj: xcodeProj))
+                }
+            }
+            try await taskGroup.waitForAll()
+            return targets
         }
```

Comment:
Is this thread-safe? multiple concurrent tasks are appending to a shared mutable array.


```suggestion
let targets = try await withThrowingTaskGroup(of: Target.self, returning: [Target].self) { taskGroup in
    for pbxTarget in pbxProject.targets {
        taskGroup.addTask {
            try await targetMapper.map(pbxTarget: pbxTarget, xcodeProj: xcodeProj)
        }
    }

    var targets: [Target] = []
    for try await target in taskGroup {
        targets.append(target)
    }
    
    return targets
}
```

Reply:
Yeah, I was kind of hoping structured concurrency would catch something like this. I do wonder if Swift makes this thread-safe under the hood, but let's go with your suggestion. That's how we implemented concurrentMap, so let's align with that.

Reply:
Yeah, you think it would throw an error with strict concurrency so it might make it safe, I am not sure.